code/modules/power/apc.dm

```diff
@@ -660,6 +662,25 @@
 				"You hear bang")*/
 			..() //Sanity
 
+/obj/machinery/power/apc/exchange_parts(mob/user, obj/item/weapon/storage/bag/gadgets/part_replacer/W)
+	if (W.bluespace || wiresexposed || opened)
+		var/obj/item/weapon/cell/max_cell
+		for(var/obj/item/weapon/cell/component in W.contents)
+			if (!max_cell || max_cell.rating < component.rating)
```

Comment:
place the right side of the or statement in parenthesis to be safe, I think this might interpret the || before the <